### PR TITLE
JAMES-3173 Document inconsistencies in admin procedures

### PR DESCRIPTION
**Pull request by LanKhuat**

### CODE REVIEW

File: src/site/markdown/server/manage-guice-distributed-james.md [hunk 1]
```diff
@@ -337,6 +340,52 @@ to sanitize his mailbox denormalization.
...
 firewall rules.
 
+### Mailboxes Counters
+
+Cassandra maintains a per mailbox projection for message count and unseen message count. Failures during the denormalization 
+process will lead to inconsistencies between the counts and the content of `imapUidTable`.
+
+#### How to detect the inconsistencies
+
+Invalid message count/message unseen count can be reported in the Mail User Agent (IMAP & JMAP).
```
Comment (chibenwa) [inline src/site/markdown/server/manage-guice-distributed-james.md:350]: Invalid **or incorrect** message count/message unseen count can be seen in Mail User Agents (IMAP or JMAP).

Comment (chibenwa) [reply src/site/markdown/server/manage-guice-distributed-james.md:350]: Invalid values are reported in the logs as warning with the following class `org.apache.james.mailbox.model.MailboxCounters`
and the following message prefix `Invalid mailbox counters`.

File: src/site/markdown/server/manage-guice-distributed-james.md [hunk 2]
```diff
@@ -337,6 +340,52 @@ to sanitize his mailbox denormalization.
...
+return the correct result.
+
+### Quotas
+
+User can monitor the amount of space and message count he is allowed to use, and that he is effectively using. James relies on 
+Event Bus and Cassandra to track the quota of an user. Upon Cassandra failure, this value can be incorrect.
+
+#### How to detect the inconsistencies
+
+
```
Comment (chibenwa) [inline src/site/markdown/server/manage-guice-distributed-james.md:365]: Here an incorrect/invalid value can be displayed in the MUA as well.

File: src/site/markdown/server/manage-guice-distributed-james.md [hunk 3]
```diff
@@ -281,6 +281,9 @@ Here is the list of known inconsistencies:
  - [RRT (RecipientRewriteTable) mapping sources](#Rrt_RecipientRewriteTable_mapping_sources)
  - [Jmap message fast view projections](#Jmap_message_fast_view_projections)
  - [Mailboxes](#Mailboxes)
+ - [Mailboxes Counters](#Mailboxes_counters)
+ - [Quotas](#Quotas)
+ - [Messages](#Messages)
```
Comment (mbaechler) [inline src/site/markdown/server/manage-guice-distributed-james.md:286]: decent ordering suggestion

```suggestion
 - [Jmap message fast view projections](#Jmap_message_fast_view_projections)
 - [Mailboxes](#Mailboxes)
 - [Mailboxes Counters](#Mailboxes_counters)
 - [Messages](#Messages)
 - [Quotas](#Quotas)
 - [RRT (RecipientRewriteTable) mapping sources](#Rrt_RecipientRewriteTable_mapping_sources)
```

File: src/site/markdown/server/manage-guice-distributed-james.md [hunk 4]
```diff
@@ -337,6 +340,53 @@ to sanitize his mailbox denormalization.
 In order to ensure being offline, stop the traffic on SMTP, JMAP and IMAP ports, for example via re-configuration or 
 firewall rules.
 
+### Mailboxes Counters
+
+Cassandra maintains a per mailbox projection for message count and unseen message count. Failures during the denormalization
```
Comment (mbaechler) [inline src/site/markdown/server/manage-guice-distributed-james.md:345]:
```suggestion
James maintains a per mailbox projection for message count and unseen message count. Failures during the denormalization 
```

File: src/site/markdown/server/manage-guice-distributed-james.md [hunk 5]
```diff
@@ -337,6 +340,53 @@ to sanitize his mailbox denormalization.
 In order to ensure being offline, stop the traffic on SMTP, JMAP and IMAP ports, for example via re-configuration or 
 firewall rules.
 
+### Mailboxes Counters
+
+Cassandra maintains a per mailbox projection for message count and unseen message count. Failures during the denormalization 
+process will lead to inconsistencies between the counts and the content of `imapUidTable`.
```
Comment (mbaechler) [inline src/site/markdown/server/manage-guice-distributed-james.md:346]: there's a lack of symmetry: either you talk only of tables or only of concepts but you should mix the two

File: src/site/markdown/server/manage-guice-distributed-james.md [hunk 6]
```diff
@@ -337,6 +340,53 @@ to sanitize his mailbox denormalization.
...
+
+#### How to detect the inconsistencies
+
+Incorrect message count/message unseen count could be seen in the `Mail User Agent` (IMAP or JMAP). Invalid values are reported in the logs 
+as warning with the following class `org.apache.james.mailbox.model.MailboxCounters` and the following message prefix: `Invalid mailbox counters`.
+
+#### How to solve
+
+Execute the [recompute Mailbox counters task](manage-webadmin.html#Recomputing mailbox counters). 
+This task is not concurrent-safe. Concurrent increments & decrements will be ignored during a single mailbox processing. Re-running this task will eventually
```
Comment (mbaechler) [inline src/site/markdown/server/manage-guice-distributed-james.md:356]:
```suggestion
This task is not concurrent-safe. Concurrent increments & decrements will be ignored during a single mailbox processing. Re-running this task may eventually 
```

File: src/site/markdown/server/manage-guice-distributed-james.md [hunk 7]
```diff
@@ -337,6 +340,53 @@ to sanitize his mailbox denormalization.
...
+#### How to solve
+
+Execute the [recompute Mailbox counters task](manage-webadmin.html#Recomputing mailbox counters). 
+This task is not concurrent-safe. Concurrent increments & decrements will be ignored during a single mailbox processing. Re-running this task will eventually 
+return the correct result.
+
+### Quotas
+
+User can monitor the amount of space and message count he is allowed to use, and that he is effectively using. James relies on 
+Event Bus and Cassandra to track the quota of an user. Upon Cassandra failure, this value can be incorrect.
```
Comment (mbaechler) [inline src/site/markdown/server/manage-guice-distributed-james.md:362]:
```suggestion
an event bus and Cassandra to track the quota of an user. Upon Cassandra failure, this value can be incorrect.
```

File: src/site/markdown/server/manage-guice-distributed-james.md [hunk 8]
```diff
@@ -337,6 +340,53 @@ to sanitize his mailbox denormalization.
...
+
+#### How to detect the inconsistencies
+
+Incorrect quotas could be seen in the `Mail User Agent` (IMAP or JMAP).
+
+#### How to solve
+
+Execute the [recompute Quotas counters task](manage-webadmin.html#Recomputing current quotas for users). 
+This task is not concurrent-safe. Concurrent operations will result in an invalid quota to be persisted. Re-running this task will
```
Comment (mbaechler) [inline src/site/markdown/server/manage-guice-distributed-james.md:371]:
```suggestion
This task is not concurrent-safe. Concurrent operations will result in an invalid quota to be persisted. Re-running this task may 
```

File: src/site/markdown/server/manage-guice-distributed-james.md [hunk 9]
```diff
@@ -337,6 +340,53 @@ to sanitize his mailbox denormalization.
...
+
+#### How to detect the inconsistencies
+
+User can see a message in JMAP but not in IMAP, or mark a message as 'SEEN' in JMAP but the message flag is still unchanged in IMAP.
+
+#### How to solve
+
+Execute the [solve Cassandra message inconsistencies task](manage-webadmin.html#Fixing_messages_inconsistencies).
+This task is not concurrent-safe. User actions concurrent to the inconsistency fixing task could result in new inconsistencies 
+being created. However the source of truth `imapUidTable` will not be affected and thus re-running this task will eventually
```
Comment (mbaechler) [inline src/site/markdown/server/manage-guice-distributed-james.md:387]:
```suggestion
being created. However the source of truth `imapUidTable` will not be affected and thus re-running this task may eventually 
```

File: src/site/markdown/server/manage-guice-distributed-james.md (hunk 3 above)
Comment (mbaechler) [reply src/site/markdown/server/manage-guice-distributed-james.md:286]: could you order sections accordingly?

File: src/site/markdown/server/manage-guice-distributed-james.md [hunk 10]
```diff
@@ -337,6 +324,69 @@ to sanitize his mailbox denormalization.
...
+#### How to solve
+
+Execute the [recompute Mailbox counters task](manage-webadmin.html#Recomputing mailbox counters). 
+This task is not concurrent-safe. Concurrent increments & decrements will be ignored during a single mailbox processing. 
+Re-running this task may eventually return the correct result.
+
+### Messages
+
+Messages are denormalized and stored in both `imapUidTable` (source of truth) and `messageIdTable`. Failure in the denormalization 
+process will cause inconsistencies between the two table.
```
Comment (Arsnael) [inline src/site/markdown/server/manage-guice-distributed-james.md:346]: s/table/tables

**Comment by LanKhuat** [comment]
Squashed & Added ticket number

**Comment by Arsnael** [comment]
Merged

**Pull request closed without merging**